search.go

```diff
@@ -571,26 +571,50 @@ func (sr *SearchResult) Size() int {
 }
 
 func (sr *SearchResult) String() string {
-	rv := ""
+	// Helper to format one hit
+	formatHit := func(i int, hit *search.DocumentMatch, start int) string {
```

Comment:
Don't see the need for this being a closure - wouldn't it be faster to declare this outside as a utility?